nixos/modules/services/web-apps/weblate.nix

```
@@ -0,0 +1,421 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.weblate;
+
+  dataDir = "/var/lib/weblate";
+  staticRoot = "${dataDir}/static";
+  compressRoot = "${dataDir}/compressor-cache";
+  settingsDir = "${dataDir}/settings";
+
+  finalPackage = cfg.package.overridePythonAttrs (old: {
+    # We only support the PostgreSQL backend in this module
+    dependencies = old.dependencies ++ cfg.package.optional-dependencies.postgres;
+    # Use a settings module in dataDir, to avoid having to rebuild the package
+    # when user changes settings.
+    makeWrapperArgs = (old.makeWrapperArgs or [ ]) ++ [
+      "--set PYTHONPATH  \"${settingsDir}\""
+      "--set DJANGO_SETTINGS_MODULE \"settings\""
+    ];
+  });
+  inherit (finalPackage) python;
+
+  pythonEnv = python.buildEnv.override {
+    extraLibs = with python.pkgs; [
+      (toPythonModule finalPackage)
+      celery
+    ];
+  };
+
+  # This extends and overrides the weblate/settings_example.py code found in upstream.
+  weblateConfig =
+    ''
+      # This was autogenerated by the NixOS module.
+
+      SITE_TITLE = "Weblate"
+      SITE_DOMAIN = "${cfg.localDomain}"
+      # TLS terminates at the reverse proxy, but this setting controls how links to weblate are generated.
+      ENABLE_HTTPS = True
+      SESSION_COOKIE_SECURE = ENABLE_HTTPS
+      DATA_DIR = "${dataDir}"
+      CACHE_DIR = f"{DATA_DIR}/cache"
+      STATIC_ROOT = "${staticRoot}"
+      MEDIA_ROOT = "/var/lib/weblate/media"
+      COMPRESS_ROOT = "${compressRoot}"
+      DEBUG = False
+
+      DATABASES = {
+        "default": {
+          "ENGINE": "django.db.backends.postgresql",
+          "HOST": "/run/postgresql",
+          "NAME": "weblate",
+          "USER": "weblate",
+        }
+      }
+
+      with open("${cfg.djangoSecretKeyFile}") as f:
+        SECRET_KEY = f.read().rstrip("\n")
+
+      CACHES = {
+        "default": {
+          "BACKEND": "django_redis.cache.RedisCache",
+          "LOCATION": "unix://${config.services.redis.servers.weblate.unixSocket}",
+          "OPTIONS": {
+              "CLIENT_CLASS": "django_redis.client.DefaultClient",
```

Maybe this is no longer necessary, but in the flake you had `PARSER_CLASS`. I wonder if you ran into the same error relating to that I did. It resolved when I prefixed `HiredisParser` with a `_`.
```suggestion
              "CLIENT_CLASS": "django_redis.client.DefaultClient",
              "PARSER_CLASS": "redis.connection._HiredisParser",
```